test/models/podcast_planner_test.rb

```diff
@@ -0,0 +1,423 @@
+            # Monday, May 21
+            DateTime.new(2001, 5, 21),
+            # Monday, June 4
+            DateTime.new(2001, 6, 4),
+            # Monday, June 18
+            DateTime.new(2001, 6, 18),
+            # Monday, July 2
+            DateTime.new(2001, 7, 2),
+            # Monday, July 16
+            DateTime.new(2001, 7, 16),
+            # Monday, August 6
+            DateTime.new(2001, 8, 6),
+            # Monday, August 20
+            DateTime.new(2001, 8, 20)
+          ]
+          planner.dates.each do |date|
+            assert_equal date > planner.start_date, true
+            assert_equal date <= planner.end_date, true
+          end
+        end
+
+        it "calculates by period if periodic" do
+          assert_nil planner.dates
+          planner.week_condition = "periodic"
+
+          # every other week
+          planner.period = 2
+          planner.generate_dates!
+          assert_equal planner.dates, [
+            # Monday, March 5
+            DateTime.new(2001, 3, 5),
+            # Monday, March 19
+            DateTime.new(2001, 3, 19),
+            # Monday, April 2
+            DateTime.new(2001, 4, 2),
+            # Monday, April 16
+            DateTime.new(2001, 4, 16),
+            # Monday, April 30
+            DateTime.new(2001, 4, 30),
+            # Monday, May 14
+            DateTime.new(2001, 5, 14),
+            # Monday, May 28
+            DateTime.new(2001, 5, 28),
+            # Monday, June 11
+            DateTime.new(2001, 6, 11),
+            # Monday, June 25
+            DateTime.new(2001, 6, 25),
+            # Monday, July 9
+            DateTime.new(2001, 7, 9),
+            # Monday, July 23
+            DateTime.new(2001, 7, 23),
+            # Monday, August 6
+            DateTime.new(2001, 8, 6),
+            # Monday, August 20
+            DateTime.new(2001, 8, 20)
+          ]
+          planner.dates.each do |date|
+            assert_equal date > planner.start_date, true
+            assert_equal date <= planner.end_date, true
+          end
+
+          # every 4 weeks
+          planner.period = 4
+          planner.generate_dates!
+          assert_equal planner.dates, [
+            # Monday, March 5
+            DateTime.new(2001, 3, 5),
+            # Monday, April 2
+            DateTime.new(2001, 4, 2),
+            # Monday, April 30
+            DateTime.new(2001, 4, 30),
+            # Monday, May 28
+            DateTime.new(2001, 5, 28),
+            # Monday, June 25
+            DateTime.new(2001, 6, 25),
+            # Monday, July 23
+            DateTime.new(2001, 7, 23),
+            # Monday, August 20
+            DateTime.new(2001, 8, 20)
+          ]
+          planner.dates.each do |date|
+            assert_equal date > planner.start_date, true
+            assert_equal date <= planner.end_date, true
+          end
+        end
+      end
+    end
+
+    describe "generate_drafts!" do
+      before do
```

Comment:
instead of using `generate_dates!` here (and having to set all of the different attributes necessary), i'm just assigning `dates` for brevity.